question/bank/managecategories/classes/question_category_list_item.php

```diff
@@ -90,15 +98,79 @@ public function item_html($extraargs = []) : string {
             $idnumber = $category->idnumber;
         }
         $questioncount = ' (' . $category->questioncount . ')';
-        $categorydesc = format_text($category->info, $category->infoformat,
-            ['context' => $this->parentlist->context, 'noclean' => true]);
+        $checked = get_user_preferences('question_bank_qbshowdescr');
+        if ($checked) {
+            $categorydesc = format_text($category->info, $category->infoformat,
+                ['context' => $this->parentlist->context, 'noclean' => true]);
+        } else {
+            $categorydesc = '';
+        }
+        $menu = new action_menu();
+        $menu->set_menu_trigger(get_string('edit'));
+        if ($this->children->editable) {
+            // Sets up edit link.
+            if (has_capability('moodle/category:manage', $context)) {
+                $thiscontext = (int)$this->item->contextid;
+                $editurl = new moodle_url('#');
+                $selector = '[data-action=editcategory-'. $category->id .']';
+                $PAGE->requires->js_call_amd('qbank_managecategories/addcategory_dialogue', 'initModal',
+                    [$selector, $thiscontext, $category->id]);
+                $menu->add(new action_menu_link(
+                    $editurl,
+                    new pix_icon('t/edit', 'edit'),
+                    get_string('editsettings'),
+                    false,
+                    ['data-action' => "editcategory-{$category->id}"]
+                ));
+                // Don't allow delete if this is the top category, or the last editable category in this context.
+                if (!helper::question_is_only_child_of_top_category_in_context($category->id)) {
+                    // Sets up delete link.
+                    $deleteurl = new moodle_url('/question/bank/managecategories/category.php',
+                        ['delete' => $category->id, 'sesskey' => sesskey()]);
+                    if ($courseid !== 0) {
+                        $deleteurl->param('courseid', $courseid);
+                    } else {
+                        $deleteurl->param('cmid', $cmid);
+                    }
+                    $menu->add(new action_menu_link(
+                        $deleteurl,
+                        new pix_icon('t/delete', 'delete'),
+                        get_string('delete'),
+                        false
+                    ));
+                }
+            }
+        }
 
-        // Don't allow delete if this is the top category, or the last editable category in this context.
-        $deleteurl = null;
-        if ($category->parent && !helper::question_is_only_child_of_top_category_in_context($category->id)) {
-            $deleteurl = new moodle_url($this->parentlist->pageurl, ['delete' => $this->id, 'sesskey' => sesskey()]);
+        // Sets up export to XML link.
+        if (class_exists('\\qbank_exportquestions\\form\\export_form')) {
```

Comment:
need the plugin enable and disable check with this class check, class is for regression issues, plugin check is for plugin enable or disable